src/urldecode.cpp

```diff
@@ -32,3 +32,23 @@ CharacterVector url_decode2(CharacterVector url) {
     return output;
   });
 }
+
+std::string str_reverse(std::string x) {
+  std::reverse(x.begin(), x.end());
+  return x;
+}
+
+//[[Rcpp::export]]
+CharacterVector url_reverse(CharacterVector urls) {
```

Comment:
I would

```cpp
CharacterVector url_reverse(const CharacterVector& urls) {
```

to prevent copy

Reply:
@chainsawriot I get error messages when doing that which I am unable to resolve 

```
 urldecode.cpp: In function ‘Rcpp::CharacterVector url_reverse(const CharacterVector&)’:
   urldecode.cpp:46:17: error: ambiguous overload for ‘operator==’ (operand types are ‘Rcpp::Vector<16>::const_Proxy’ {aka ‘Rcpp::internal::const_string_proxy<16>’} and ‘SEXP’ {aka ‘SEXPREC*’})
```

Reply:
Oops I didn't see there is a direct comparison of element below and therefore cannot use `const`.
This is enough to prevent copying

```r
CharacterVector url_reverse(CharacterVector& urls) {
```

it works on my machine™

Reply:
If you like, you can also change

```cpp
CharacterVector url_decode2(const CharacterVector& url) {
```

Checked.